src/utils/parallelPaginatedRequest.js

```diff
@@ -0,0 +1,75 @@
+import axios from 'axios';
+
+async function fetchPage(apiUrl, requestConfig, page, pageSize, maxRetries) {
```

Comment:
nit: we're pretty consistent throughout the repo with the `const fetchPage = async (...whatever)` convention. Why go `async function` instead?